Linking in local Backstage packages

<!--- Provide a general summary of the issue in the Title above -->
I have created my backstage app. So, I need to work on @backstage/core package located in another folder.
by following this instructions (https://backstage.io/docs/getting-started/create-an-app#linking-in-local-backstage-packages), it is installed successfully. 

## Expected Behavior

<!--- Tell us what should happen -->
With this in place you I now modify the @backstage/core package within the main repo, and have those changes be reflected and tested in my app. Simply run your app using yarn start as normal.

## Current Behavior

<!--- Tell us what happens instead of the expected behavior -->
It is installed @backstage/core from registry npm, but the local backstage package components are not installed instead.

## Steps to Reproduce

<!--- Provide a link to a live example, or an unambiguous set of steps to -->
<!--- reproduce this bug. Include code or configuration to reproduce, if relevant -->

1. Add path to workspace files **package.json** and **lerna.json** 
```
"workspaces": {
    "packages": [
      "packages/*",
      "plugins/*",
      "../backstage/packages/core"  // -> new path added
    ]
  },
```
2. Install command:
```
yarn install
```
Also, I deleted the root **node_modules** and reinstall as additional step.

## Context

I am trying to get custom componets for my custom application.
<!--- How has this issue affected you? What are you trying to accomplish? -->
<!--- Providing context (e.g. links to configuration settings, -->
<!--- stack trace or log data) helps us come up with a solution that is most useful in the real world -->
On the other hand, if I run the command:
```
yarn tsc
```
I get stack trace error log: 

```
Type 'import("/backstage-sample/sample/node_modules/@backstage/theme/dist/index").BackstageTheme' is not assignable to type 'import("/backstage/packages/theme/src/types").BackstageTheme'.
...
packages/app/src/sidebar.tsx:29:18 - error TS2322: Type 'OverridableComponent<SvgIconTypeMap<{}, "svg">>' is not assignable to type 'ComponentType<OverrideProps<SvgIconTypeMap<{}, "svg">, "svg">>'.
  Type 'OverridableComponent<SvgIconTypeMap<{}, "svg">>' is not assignable to type 'FunctionComponent<OverrideProps<SvgIconTypeMap<{}, "svg">, "svg">>'.
    Types of parameters 'props' and 'props' are incompatible.
```

I need help to solve this issue. Any solution?

## Your Environment

<!--- Include as many relevant details about the environment you experienced the bug in -->

- NodeJS Version (v14): v14
- Operating System and Version (e.g. Ubuntu 14.04): MacOS 10.15.7
- Browser Information: Safari, Chrome, (it is not important)

--- a/docs/getting-started/create-an-app.md
+++ b/docs/getting-started/create-an-app.md
@@ -74,6 +74,22 @@ those plugins in your backend. This is because the transformation of backend
 module tree stops whenever a non-local package is encountered, and from that
 point node will `require` packages directly for that entire module subtree.
 
+Type checking can also have issues when linking in external packages, since the
+linked in packages will use the types in the external project and dependency
+version mismatches between the two projects may cause errors. To fix any of
+those errors you need to sync versions of the dependencies in the two projects.
+A simple way to do this can be to copy over `yarn.lock` from the external
+project and run `yarn install`, although this is quite intrusive and can cause
+other issues in existing projects, so use this method with care. It can often be
+best to simply ignore the type errors, as app serving will work just fine
+anyway.
+
+Another issue with type checking is that the incremental type cache doesn't
+invalidate correctly for the linked in packages, causing type checking to not
+reflect changes made to types. You can work around this by either setting
+`compilerOptions.incremental = false` in `tsconfig.json`, or by deleting the
+types cache folder `dist-types` before running `yarn tsc`.
+
 ### Troubleshooting
 
 The create app command doesn't always work as expected, this is a collection of

